python/openai/openai_frontend/main.py

```diff
@@ -73,21 +86,33 @@ def parse_args():
     return parser.parse_args()
 
 
-if __name__ == "__main__":
+def main():
     args = parse_args()
-    # NOTE: configurations can be passed to FastAPI app through a builder
-    # function like init_app in future, but using env vars for simplicity.
-    os.environ["TRITON_MODEL_REPOSITORY"] = args.model_repository
-    if args.tokenizer:
-        os.environ["TOKENIZER"] = args.tokenizer
-
-    os.environ["TRITON_LOG_VERBOSE_LEVEL"] = str(args.tritonserver_log_level)
-
-    app = init_app()
-    uvicorn.run(
-        app,
-        host=args.host,
-        port=args.port,
-        log_level=args.uvicorn_log_level,
-        timeout_keep_alive=5,
+
+    # Initialize a Triton Inference Server pointing at LLM models
+    server: tritonserver.Server = tritonserver.Server(
+        model_repository=args.model_repository,
+        log_verbose=args.tritonserver_log_verbose_level,
+        log_info=True,
+        log_warn=True,
+        log_error=True,
+    ).start(wait_until_ready=True)
+
+    # Wrap Triton Inference Server in an interface-conforming "LLMEngine"
+    engine: TritonLLMEngine = TritonLLMEngine(server=server, tokenizer=args.tokenizer)
+
+    # Attach TritonLLMEngine as the backbone for inference and model management
+    frontend: TritonOpenAIFrontend = TritonOpenAIFrontend(
+        host=args.host, port=args.port, log_level=args.uvicorn_log_level, engine=engine
     )
+
+    # Gracefully shutdown when receiving signals for testing and interactive use
+    signal.signal(signal.SIGINT, partial(signal_handler, server, frontend))
+    signal.signal(signal.SIGTERM, partial(signal_handler, server, frontend))
+
+    # Blocking call until killed or interrupted with SIGINT
+    frontend.start()
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
Is this something common that the entry point logic is wrap into another function?

Reply:
Relatively common:
- https://github.com/vllm-project/vllm/blob/9c71c97ae24ae6f5209a475149808e25554cfe99/vllm/scripts.py#L161-L162
- https://github.com/fastapi/fastapi/blob/9416e89bd7e3491157f8c962012aebd7be7ed5d6/scripts/deploy_docs_status.py#L106

Primarily just makes it so that this function could be imported from another script to be run directly rather than launching the script as a subprocess or something. I'm not taking advantage of this, so it makes no difference one way or the other currently.